controls/tests.py

```diff
@@ -293,19 +288,18 @@ def test_component_import_oscal_json(self):
         # but we can't test individual statements because the UUIDs are randomly generated and not consistent
         # with the OSCAL JSON file. So we simply do a count.
 
-        var_sleep(3) # Needed to allow page to refresh and messages to render
-
         # Test that duplicate Components are re-imported with a different name and that Statements get reimported
-        self.click_element('a#component-import-oscal')
+
+        wait_for_sleep_after(lambda: self.click_element('a#component-import-oscal'))
```

Comment:
This seems to be a recurring question. Here, the wait should happen before this click. Instead, it seems like the element is being clicked and _then_ we're waiting for the wait. If this is the correct use of the function, maybe it just needs to be renamed to "wait_for_sleep_before", or maybe it is a logic issue and the call to the wait function needs to be on the function before the initial var_sleep.

Reply:
So if self.click_element can't find the element to click this should fail the first time. If it does the wait_for_sleep_after func will recall this on that element for up to 10seconds(max_wait). So it is not clicking several times, it is instead _attempting_ to click the element however many times.

Reply:
Okay great, that makes sense. So it's really a naming misunderstanding on my part. I was thinking the function said "wait for sleep _after_ doing this thing". But it's really saying "wait for sleep, then do this thing afterwards". See my previous naming suggestions to maybe clarify this function, but the actual logic of it seems good to me.